### PR TITLE
[youtube] update invidious instance list

### PR DESCRIPTION
**Pull request by B0pol**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Updated the invidious instance list using https://github.com/iv-org/documentation/blob/master/Invidious-Instances.md

### CODE REVIEW

**Comment by pukkandan** [comment]
Something like https://github.com/yt-dlp/yt-dlp/commit/df0c81513e0bb37986d00c532a5ad8cef31a24ea would be more beneficial than updating the list periodically

Ref: https://github.com/iv-org/invidious/pull/1730

File: youtube_dl/extractor/youtube.py [hunk 1]
```diff
@@ -353,7 +353,7 @@ class YoutubeIE(YoutubeBaseInfoExtractor):
         r'(?:www\.)?invidious\.13ad\.de',
         r'(?:www\.)?invidious\.mastodon\.host',
         r'(?:www\.)?invidious\.zapashcanon\.fr',
-        r'(?:www\.)?invidious\.kavin\.rocks',
+        r'(?:www\.)?(invidious(-us)?|piped)\.kavin\.rocks',
```
Comment (dstftw) [inline youtube_dl/extractor/youtube.py:356]: Do not capture groups you don't use.

**Review by dstftw — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/youtube.py (hunk 1 above)
Comment (B0pol) [reply youtube_dl/extractor/youtube.py:356]: ?

Comment (pukkandan) [reply youtube_dl/extractor/youtube.py:356]: He means you are supposed to make it a non-capturing group. Like this:
`(?:invidious(?:-us)?|piped)`

**Pull request merged**